### PR TITLE
Template set grpc only (other dependencies already merged)

### PR DESCRIPTION
**Pull request by moljac**
### Describe your contribution

1. moved `io.grpc.*` and `com.google.gson` bindings 
- `io.grpc:grpc-android` -  -> 1.28.0
- `io.grpc:grpc-api` -  -> 1.28.0
- `com.google.code.gson:gson` -  -> 2.8.8
  
2. created template sets

## Updated/new (moved) artifacts

- `io.grpc:grpc-android` -  -> 1.28.0
- `io.grpc:grpc-api` -  -> 1.28.0
- `com.google.code.gson:gson` -  -> 2.8.8

### Does this change any of the generated binding API's?

No.

`net6-android` TFM added

### CODE REVIEW

**Comment by jpobst** [comment]
The currently published versions on NuGet are `1.37.0`.  Let's switch this to match that version, which should reduce the amount of diffs to verify.

**Comment by moljac** [comment]
> The currently published versions of GRPC on NuGet are `1.37.0`. Let's switch this to match that version, which should reduce the amount of diffs to verify.

Tried that:

```
./template-sets-grpc-only/generated/io.grpc.grpc-stub/io.grpc.grpc-stub.csproj : error NU1102: Unable to find package Xamarin.Google.Guava with version (>= 30.0.0) [./template-sets-grpc-only/generated/GooglePlayServices.sln]
./template-sets-grpc-only/generated/io.grpc.grpc-stub/io.grpc.grpc-stub.csproj : error NU1102:   - Found 9 version(s) in nuget.org [ Nearest version: 29.0.0 ] [./template-sets-grpc-only/generated/GooglePlayServices.sln]
./template-sets-grpc-only/generated/io.grpc.grpc-stub/io.grpc.grpc-stub.csproj : error NU1102:   - Found 0 version(s) in /usr/local/share/dotnet/library-packs [./template-sets-grpc-only/generated/GooglePlayServices.sln]
./template-sets-grpc-only/generated/io.grpc.grpc-api/io.grpc.grpc-api.csproj : error NU1102: Unable to find package Xamarin.Google.Guava with version (>= 30.0.0) [./template-sets-grpc-only/generated/GooglePlayServices.sln]
./template-sets-grpc-only/generated/io.grpc.grpc-api/io.grpc.grpc-api.csproj : error NU1102:   - Found 9 version(s) in nuget.org [ Nearest version: 29.0.0 ] [./template-sets-grpc-only/generated/GooglePlayServices.sln]
./template-sets-grpc-only/generated/io.grpc.grpc-api/io.grpc.grpc-api.csproj : error NU1102:   - Found 0 version(s) in /usr/local/share/dotnet/library-packs [./template-sets-grpc-only/generated/GooglePlayServices.sln]
./template-sets-grpc-only/generated/io.grpc.grpc-protobuf-lite/io.grpc.grpc-protobuf-lite.csproj : error NU1102: Unable to find package Xamarin.Google.Guava with version (>= 30.0.0) [./template-sets-grpc-only/generated/GooglePlayServices.sln]
./template-sets-grpc-only/generated/io.grpc.grpc-protobuf-lite/io.grpc.grpc-protobuf-lite.csproj : error NU1102:   - Found 9 version(s) in nuget.org [ Nearest version: 29.0.0 ] [./template-sets-grpc-only/generated/GooglePlayServices.sln]

```

With:

```
      {
        "groupId": "com.google.guava",
        "artifactId": "guava",
        "version": "30.0-android",
        "nugetVersion": "30.0.0",
        "nugetId": "Xamarin.Google.Guava",
        "dependencyOnly": true
      },
```
The error is:

```
Dependency errors : 6
1
	System.Exception: Artifact 'io.grpc.grpc-protobuf-lite' has unknown 'Runtime' dependency 'org.codehaus.mojo.animal-sniffer-annotations'. Either fulfill or exclude this dependency.
2
	System.Exception: Artifact 'io.grpc.grpc-stub' has unknown 'Runtime' dependency 'org.codehaus.mojo.animal-sniffer-annotations'. Either fulfill or exclude this dependency.
3
	System.Exception: Artifact 'io.grpc.grpc-okhttp' has unknown 'Runtime' dependency 'org.codehaus.mojo.animal-sniffer-annotations'. Either fulfill or exclude this dependency.
4
	System.Exception: Artifact 'io.grpc.grpc-core' has unknown 'Runtime' dependency 'org.codehaus.mojo.animal-sniffer-annotations'. Either fulfill or exclude this dependency.
5
	System.Exception: Artifact 'io.grpc.grpc-android' has unknown 'Runtime' dependency 'org.codehaus.mojo.animal-sniffer-annotations'. Either fulfill or exclude this dependency.
6
	System.Exception: Artifact 'io.grpc.grpc-api' has unknown 'Runtime' dependency 'org.codehaus.mojo.animal-sniffer-annotations'. Either fulfill or exclude this dependency.

```

Adding

```
        "excludedRuntimeDependencies": "org.codehaus.mojo.animal-sniffer-annotations",
```

leads to:

```
./template-sets-grpc-only/generated/io.grpc.grpc-stub/io.grpc.grpc-stub.csproj : error NU1102: Unable to find package Xamarin.Google.Guava with version (>= 30.0.0) [./template-sets-grpc-only/generated/GooglePlayServices.sln]
./template-sets-grpc-only/generated/io.grpc.grpc-stub/io.grpc.grpc-stub.csproj : error NU1102:   - Found 9 version(s) in nuget.org [ Nearest version: 29.0.0 ] [./template-sets-grpc-only/generated/GooglePlayServices.sln]
./template-sets-grpc-only/generated/io.grpc.grpc-stub/io.grpc.grpc-stub.csproj : error NU1102:   - Found 0 version(s) in /usr/local/share/dotnet/library-packs [./template-sets-grpc-only/generated/GooglePlayServices.sln]
./template-sets-grpc-only/generated/io.grpc.grpc-api/io.grpc.grpc-api.csproj : error NU1102: Unable to find package Xamarin.Google.Guava with version (>= 30.0.0) [./template-sets-grpc-only/generated/GooglePlayServices.sln]
./template-sets-grpc-only/generated/io.grpc.grpc-api/io.grpc.grpc-api.csproj : error NU1102:   - Found 9 version(s) in nuget.org [ Nearest version: 29.0.0 ] [./template-sets-grpc-only/generated/GooglePlayServices.sln]
./template-sets-grpc-only/generated/io.grpc.grpc-api/io.grpc.grpc-api.csproj : error NU1102:   - Found 0 version(s) in /usr/local/share/dotnet/library-packs [./template-sets-grpc-only/generated/GooglePlayServices.sln]
./template-sets-grpc-only/generated/io.grpc.grpc-protobuf-lite/io.grpc.grpc-protobuf-lite.csproj : error NU1102: Unable to find package Xamarin.Google.Guava with version (>= 30.0.0) [./template-sets-grpc-only/generated/GooglePlayServices.sln]
./template-sets-grpc-only/generated/io.grpc.grpc-protobuf-lite/io.grpc.grpc-protobuf-lite.csproj : error NU1102:   - Found 9 version(s) in nuget.org [ Nearest version: 29.0.0 ] [./template-sets-grpc-only/generated/GooglePlayServices.sln]
```

Basically stuck without 2 or 3 rounds of publishing packets.

File: source/com.google.firebase/firebase-firestore/Transforms/Metadata.xml [hunk 1]
```diff
@@ -266,8 +266,25 @@
         >
         Firebase.Firestore.Model.Value.FieldValue
     </attr>
-        
-        
+    <attr
+        path="/api/package[@name='com.google.firestore.v1']/class[@name='FirestoreGrpc.FirestoreStub']/method[@name='build' and count(parameter)=2 and parameter[1][@type='io.grpc.Channel'] and parameter[2][@type='io.grpc.CallOptions']]"
```
Comment (jpobst) [inline source/com.google.firebase/firebase-firestore/Transforms/Metadata.xml:270]: What error does the new `metadata` changes fix?  It looks like they worked previously without these changes, and changing them will cause API breakages:

(same for inappmessaging)

#### Type Changed: Google.Firestore.V1.FirestoreGrpc.FirestoreBlockingStub

Removed method:

```csharp
protected FirestoreGrpc.FirestoreBlockingStub Build (Xamarin.Grpc.Channel channel, Xamarin.Grpc.CallOptions callOptions);
```

Added method:

```csharp
protected override Java.Lang.Object Build (Xamarin.Grpc.Channel channel, Xamarin.Grpc.CallOptions callOptions);
```


#### Type Changed: Google.Firestore.V1.FirestoreGrpc.FirestoreFutureStub

Removed method:

```csharp
protected FirestoreGrpc.FirestoreFutureStub Build (Xamarin.Grpc.Channel channel, Xamarin.Grpc.CallOptions callOptions);
```

Added method:

```csharp
protected override Java.Lang.Object Build (Xamarin.Grpc.Channel channel, Xamarin.Grpc.CallOptions callOptions);
```


#### Type Changed: Google.Firestore.V1.FirestoreGrpc.FirestoreStub

Removed method:

```csharp
protected FirestoreGrpc.FirestoreStub Build (Xamarin.Grpc.Channel channel, Xamarin.Grpc.CallOptions callOptions);
```

Added method:

```csharp
protected override Java.Lang.Object Build (Xamarin.Grpc.Channel channel, Xamarin.Grpc.CallOptions callOptions);
```

**Comment by jpobst** [comment]
> 1.37.0 needs newest guava.

👍 makes sense

File: source/com.google.firebase/firebase-firestore/Transforms/Metadata.xml (hunk 1 above)
Comment (moljac) [reply source/com.google.firebase/firebase-firestore/Transforms/Metadata.xml:270]: without this:

```
    <attr
        path="/api/package[@name='com.google.firestore.v1']/class[@name='FirestoreGrpc.FirestoreStub']/method[@name='build' and count(parameter)=2 and parameter[1][@type='io.grpc.Channel'] and parameter[2][@type='io.grpc.CallOptions']]"
        name="managedReturn"
        >
        Java.Lang.Object
    </attr>
```

the error is:

```
./generated/com.google.firebase.firebase-firestore/obj/Release/monoandroid90/generated/src/Google.Firestore.V1.FirestoreGrpc.cs(941,31): error CS0534: 'FirestoreGrpc.FirestoreStub' does not implement inherited abstract member 'AbstractStub.Build(Channel, CallOptions)' [./generated/com.google.firebase.firebase-firestore/com.google.firebase.firebase-firestore.csproj]
```

Comment (moljac) [reply source/com.google.firebase/firebase-firestore/Transforms/Metadata.xml:270]: besides those methods are `protected`, so API change is not that critical.

Comment (jpobst) [reply source/com.google.firebase/firebase-firestore/Transforms/Metadata.xml:270]: The Java is:
```java
public abstract class AbstractStub<S extends AbstractStub<S>> {
  protected abstract S build(Channel paramChannel, CallOptions paramCallOptions);
}
```

Looks like previously this method was not being bound, and now it is, likely due to a `generator` fix.

```csharp
[Register("build", "(Lio/grpc/Channel;Lio/grpc/CallOptions;)Lio/grpc/stub/AbstractStub;", "GetBuild_Lio_grpc_Channel_Lio_grpc_CallOptions_Handler")]
protected abstract Object Build(Channel p0, CallOptions p1);
```

I guess the breakage is ok, like you said it's `protected` API.

Comment (moljac) [reply source/com.google.firebase/firebase-firestore/Transforms/Metadata.xml:270]: > Looks like previously this method was not being bound, and now it is, likely due to a `generator` fix.

Yeah. I suspected that, but wasn't sure

Comment (moljac) [reply source/com.google.firebase/firebase-firestore/Transforms/Metadata.xml:270]: ... and yes nobody touched these bindings for quite some time (and not gone deeper into analysis)

**Review by jpobst — APPROVED** [review]
LGTM

**Pull request merged**